File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
            HashMap <Integer,Integer> numMap = new HashMap();
            for (int i =0 ; i < nums.length ; i++) {
                if(numMap.get(nums[i]) != null){
                    return true;
                }
                numMap.put(nums[i],1);
            }
            return false;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.kt
class Solution {
    fun topKFrequent(nums: IntArray, k: Int): IntArray {

        // Step 1: Count frequency
        val freqMap = HashMap<Int, Int>()
        for (num in nums) {
            freqMap[num] = freqMap.getOrDefault(num, 0) + 1
        }

        // Step 2: Create buckets
        val buckets = Array(nums.size + 1) { mutableListOf<Int>() }

        for ((num, freq) in freqMap) {
            buckets[freq].add(num)
        }

        // Step 3: Collect top k frequent
        val result = mutableListOf<Int>()

        for (i in buckets.size - 1 downTo 0) {
            for (num in buckets[i]) {
                result.add(num)
                if (result.size == k) {
                    return result.toIntArray()
                }
            }
        }

        return result.toIntArray()
    }
}
